src/components/panningViews/panToDismissView.js

```diff
@@ -0,0 +1,301 @@
+    ) {
+      this.onSwipe(swipeDirections);
+    }
+  }
+
+  onLayout = event => {
+    if (_.isUndefined(this.height)) {
+      const layout = event.nativeEvent.layout;
+      this.height = layout.height;
+      this.thresholdY = layout.height / 2;
+      this.width = layout.width;
+      this.thresholdX = layout.width / 2;
+      const {style} = this.props;
+      this.originalLeft = _.get(style, 'left', 0);
+      this.originalTop = _.get(style, 'top', 0);
+      this.initPositions();
+    }
+  };
+
+  initPositions = () => {
+    this.setNativeProps(this.originalLeft, this.originalTop);
+    this.setState({
+      deltaX: new Animated.Value(this.originalLeft),
+      deltaY: new Animated.Value(this.originalTop),
+    });
+  };
+
+  onPanStart = () => {
+    this.swipe = [undefined, undefined];
+  };
+
+  onDrag = deltas => {
+    const left = deltas[0] ? Math.round(deltas[0]) : this.originalLeft;
+    const top = deltas[1] ? Math.round(deltas[1]) : this.originalTop;
+    this.setNativeProps(left, top);
+  };
+
+  setNativeProps = (left, top) => {
+    if (this.ref.current) {
+      this.ref.current.setNativeProps({style: {left, top}});
+      this.left = left;
+      this.top = top;
+    }
+  };
+
+  onSwipe = swipeDirections => {
+    this.swipe = swipeDirections;
+  };
+
+  onPanEnd = () => {
+    const {directions} = this.getThemeProps();
+    if (this.swipe[0] || this.swipe[1]) {
+      const {isRight, isDown} = this.getDismissAnimationDirection();
+      this.animateDismiss(isRight, isDown);
+    } else {
+      const endValue = [Math.round(this.left), Math.round(this.top)];
+      if (
+        (directions.includes(PanningProvider.Directions.LEFT) && endValue[0] <= -this.thresholdX) ||
+        (directions.includes(PanningProvider.Directions.RIGHT) && endValue[0] >= this.thresholdX) ||
+        (directions.includes(PanningProvider.Directions.UP) && endValue[1] <= -this.thresholdY) ||
+        (directions.includes(PanningProvider.Directions.DOWN) && endValue[1] >= this.thresholdY)
+      ) {
+        const {isRight, isDown} = this.getDismissAnimationDirection();
+        this.animateDismiss(isRight, isDown);
+      } else {
+        this.animateToInitialPosition();
+      }
+    }
+  };
+
+  animateToInitialPosition = () => {
+    const {deltaX, deltaY} = this.state;
+    const {animationSpeed, animationBounciness} = this.props;
+    const toX = -this.left;
+    const toY = -this.top;
+    const animations = [];
+    if (!_.isUndefined(toX)) {
+      animations.push(
+        Animated.spring(deltaX, {
+          toValue: Math.round(toX),
+          speed: animationSpeed,
+          bounciness: animationBounciness,
+        }),
+      );
+    }
+
+    if (!_.isUndefined(toY)) {
+      animations.push(
+        Animated.spring(deltaY, {
+          toValue: Math.round(toY),
+          speed: animationSpeed,
+          bounciness: animationBounciness,
+        }),
+      );
+    }
+
+    this.setState({shouldAnimate: true}, () => {
```

Comment:
isn't `shouldAnimate` be called `isAnimating` or something similar? 
doesn't `shouldAnimate` represent when the animation is happening?

Reply:
Yes, oops... Done.